### PR TITLE
Added ExceltoPdf Fragment

### LINKED ISSUE #186: [New Feature] Convert excel to PDF
### What to do?
- Add a new option in the navigation drawer `Excel to PDF`
- In that fragment, allow users to select an existing excel file
- Ask for the output PDF name

### Reference:
http://thinktibits.blogspot.com/2012/12/POI-iText-Convert-XLS-to-PDF-Java-Program.html

### PR DESCRIPTION
**Pull request by jddeep**
# Description
Added the Excel to Pdf Fragment.

Screenshot_20190401-180643

Screenshot_20190401-180717
Screenshot_20190401-180741

Fixes #186 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by jddeep** [comment]
@vyankatesh24 @Swati4star Please review. A NullpointerException is thrown while creating the ExceltoPdf Fragment. It seems that the `MorphingButtons` are not getting binded using `butterknife` . Please help.

**Comment by jddeep** [comment]
Otherwise all things seem fine. Thanks.

**Comment by sidhuparas** [comment]
It's not even getting compiled. Have you checked?
`Cause: unable to find valid certification path to requested target`

**Comment by jddeep** [comment]
@sidhuparas Please check your branch or your local machine. The app compiles, builds, runs fine as you can see by the successful travis build.

**Comment by sidhuparas** [comment]
@jddeep Yes. Requesting #vyankteshberde and @srv-twry to have a look.

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    private final int mFileSelectCode = 0;
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootview = inflater.inflate(R.layout.fragment_excel_to_pdf, container,
+                false);
+        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(mActivity);
+        mMorphButtonUtility = new MorphButtonUtility(mActivity);
+        ButterKnife.bind(rootview);
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: Use `ButterKnife.bind(this, rootview);`

File: app/build.gradle [hunk 2]
```diff
@@ -28,12 +28,28 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+    packagingOptions {
```
Comment (vyankatesh24) [inline app/build.gradle]: These are "add drive storage" dependencies remove them from this PR.

File: app/build.gradle [hunk 3]
```diff
@@ -28,12 +28,28 @@ android {
...
+    }
 }
 repositories {
     mavenCentral()
-    maven { url 'https://jitpack.io' }
+    maven { url 'https://jitpack.io'}
+    maven {
+        url "https://repository.aspose.com/repo/" }
+}
+configurations {
```
Comment (vyankatesh24) [inline app/build.gradle]: These are "add drive storage" dependencies remove them from this PR.

File: app/build.gradle [hunk 4]
```diff
@@ -49,6 +65,16 @@ dependencies {
     implementation 'com.android.support:cardview-v7:28.0.0'
     implementation 'com.android.support:support-v4:28.0.0'
     implementation 'com.android.support:preference-v7:28.0.0'
+    implementation 'com.aspose:aspose-cells:19.3'
+    implementation 'com.google.android.gms:play-services-auth:16.0.1'
```
Comment (vyankatesh24) [inline app/build.gradle]: These are "add drive storage" dependencies remove them from this PR.

**Comment by jddeep** [comment]
@vyankatesh24 Please review, I have made the changes you asked. But there is this error after clicking on convert to Pdf button : 
``` swati4star.createpdf E/Error: content://com.android.externalstorage.documents/document/primary%3ACS%20GST%202018.xlsx
swati4star.createpdf E/Error message: java.io.FileNotFoundException: Can't find file: content://com.android.externalstorage.documents/document/primary%3ACS%20GST%202018.xlsx. ```
The first line shows the Uri of the excel file. But the same uri is not being able to use as a new `Workbook` as it shows `FileNotFoundException`. @Swati4star @vyankatesh24 Please help.

**Comment by jddeep** [comment]
``` swati4star.createpdf E/Error message: java.io.FileNotFoundException: Can't find file: content://com.android.externalstorage.documents/document/primary%3ACS%20GST%202018.xlsx.

**Comment by jddeep** [comment]
> `swati4star.createpdf E/Error: content://com.android.externalstorage.documents/document/primary%3ACS%20GST%202018.xlsx`

The first line shows the Uri of the excel file. But the same uri is not being able to use as a new `Workbook` as it shows `FileNotFoundException`. @Swati4star @vyankatesh24 Please help.

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 5]
```diff
@@ -309,38 +310,38 @@ public void onBackPressed() {
...
     private void closeFragmentBottomSheet() {
-        if ( mCurrentFragment instanceof InvertPdfFragment)
+        if (mCurrentFragment instanceof InvertPdfFragment)
             ((InvertPdfFragment) mCurrentFragment).closeBottomSheet();
 
         if (mCurrentFragment instanceof MergeFilesFragment)
             ((MergeFilesFragment) mCurrentFragment).closeBottomSheet();
 
-        if (mCurrentFragment instanceof RemoveDuplicatePagesFragment )
+        if (mCurrentFragment instanceof RemoveDuplicatePagesFragment)
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Why so many unrelated changes?

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+    private SharedPreferences mSharedPreferences;
+    private MorphButtonUtility mMorphButtonUtility;
+    private static final int PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT = 1;
+    private boolean mPermissionGranted = false;
+    private int mButtonClicked = 0;
+    private final int mFileSelectCode = 0;
+
+    public ExceltoPdfFragment() {
+    }
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: What's the point of this blank constructor? Java automatically creates a constructor without any arguments in case you don't have one. So you can safely remove this.

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+        }
+    }
+
+    private String trimExtension(String mExcelFileUri) {
+        String trimmed = "";
+        for (int i = 0; i < mExcelFileUri.length() - 1; i++) {
+            trimmed += mExcelFileUri.charAt(i);
+        }
+        return trimmed;
+    }
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: Could you explain what you're doing here?

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java (hunk 5 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: These were auto generated after reformatting files. Will remove.

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java (hunk 6 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: Yes will remove.

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java (hunk 7 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: This function was meant to trim the extension to get only the file name. But this is useless now. So will remove it. @srv-twry Please help in the functioning of the app. I have given error logs. Please help me out with that.

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java (hunk 6 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: But before that please help me with the functioning of the app. I shall make a final push removing these formatting changes.

**Comment by srv-twry** [comment]
@jddeep Sorry, I have seen your problem and can't quite figure out the issue. @sidhuparas @vyankatesh24 Can you figure out the problem that @jddeep is facing?

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java (hunk 6 above)
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: Do submit a push. It becomes easier to review PR if it contains only the required changes

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+    @BindView(R.id.tv_excel_file_name_bottom)
+    TextView mTextView;
+    @BindView(R.id.create_excel_to_pdf)
+    MorphingButton mCreateExcelPdf;
+
+    private SharedPreferences mSharedPreferences;
+    private MorphButtonUtility mMorphButtonUtility;
+    private static final int PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT = 1;
+    private boolean mPermissionGranted = false;
+    private int mButtonClicked = 0;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: use boolean for this

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+                        mFileSelectCode);
+            } catch (android.content.ActivityNotFoundException ex) {
+                showSnackbar(mActivity, R.string.install_file_manager);
+            }
+            mButtonClicked = 1;
+        }
+    }
+
+    @OnClick(R.id.create_excel_to_pdf)
+    public void openExcelToPdf() {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: add comments to newly added functions

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        mButtonClicked = 0;
+        switch (requestCode) {
+            case mFileSelectCode:
+                if (resultCode == RESULT_OK) {
+                    mExcelFileUri = data.getData();
+                    showSnackbar(mActivity, "Excel file selected");
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: dont use strings directly in code. they should come from strings.xml file

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+                        if (fileName.endsWith(Constants.excelExtension))
+                            mFileExtension = Constants.excelExtension;
+                        else if (fileName.endsWith(Constants.excelWorkbookExtension))
+                            mFileExtension = Constants.excelWorkbookExtension;
+                        else {
+                            showSnackbar(mActivity, R.string.extension_not_supported);
+                            return;
+                        }
+                    }
+                    fileName = "Excel file: " + fileName;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: we can have the pdf file name same as that of excel file.

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java (hunk 7 above)
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: add these type of functions in utils class

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java [hunk 12]
```diff
@@ -0,0 +1,218 @@
...
+
+    private String trimExtension(String mExcelFileUri) {
+        String trimmed = "";
+        for (int i = 0; i < mExcelFileUri.length() - 1; i++) {
+            trimmed += mExcelFileUri.charAt(i);
+        }
+        return trimmed;
+    }
+
+    private void convertToPdf(String mFilename) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: add comment

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java [hunk 13]
```diff
@@ -0,0 +1,218 @@
...
+                .show();
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        mButtonClicked = 0;
+        switch (requestCode) {
+            case mFileSelectCode:
+                if (resultCode == RESULT_OK) {
+                    mExcelFileUri = data.getData();
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: Instead of uri, try this using this to get file path
https://github.com/Swati4star/Images-to-PDF/blob/7ef993aa2dac8628d5162fcd10f8c58d75d0ccac/app/src/main/java/swati4star/createpdf/fragment/ZipToPdfFragment.java#L92

Try and see if this fixes the issue

File: app/src/main/res/menu/activity_main_drawer.xml [hunk 14]
```diff
@@ -20,6 +20,9 @@
                         android:id="@+id/nav_qrcode"
                         android:icon="@drawable/ic_qrcode_24dp"
                         android:title="@string/qr_barcode_pdf" />
+                    <item
+                        android:id="@+id/nav_excel_to_pdf"
+                        android:title="Excel to Pdf" />
```
Comment (Swati4star) [inline app/src/main/res/menu/activity_main_drawer.xml]: dont use constant strings

File: app/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -78,6 +78,7 @@
     </string-array>
 
     <string name="pdf_ext" translatable="false">.pdf</string>
+    <string name="xls_ext" translatable="false">.xls</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: why do you need this twice? You have already defined this value in constants file also
excelExtension

File: app/src/main/res/values-es/strings.xml [hunk 16]
```diff
@@ -400,5 +402,7 @@
     <string name="images_to_pdf">Images to PDF</string>
   
     <string name="font_color">Font Color</string>
+
+    <string name="drive_selection">Select from Drive</string>
```
Comment (vyankatesh24) [inline app/src/main/res/values-es/strings.xml]: Why this string?

File: app/src/main/res/values/strings.xml [hunk 17]
```diff
@@ -499,4 +501,9 @@
     <string name="zip_to_pdf">ZIP to PDF</string>
     <string name="convert_to_pdf">convert to PDF</string>
     <string name="error_no_image_in_zip">ZIP file contains no image</string>
+
+    <string name="drive_selection">Select from Drive</string>
+
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (vyankatesh24) [inline app/src/main/res/values/strings.xml]: Why this string?

File: app/src/main/res/layout/fragment_excelto_pdf.xml [hunk 18]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (vyankatesh24) [inline app/src/main/res/layout/fragment_excelto_pdf.xml]: Why these two layout XMLs?

File: app/build.gradle [hunk 19]
```diff
@@ -32,8 +32,10 @@ android {
 repositories {
     mavenCentral()
     maven { url 'https://jitpack.io' }
+    maven {
+        url "https://repository.aspose.com/repo/"
+    }
 }
-
```
Comment (vyankatesh24) [inline app/build.gradle]: Don't remove this line!

**Comment by vyankatesh24** [comment]
> > `swati4star.createpdf E/Error: content://com.android.externalstorage.documents/document/primary%3ACS%20GST%202018.xlsx`
> 
> The first line shows the Uri of the excel file. But the same uri is not being able to use as a new `Workbook` as it shows `FileNotFoundException`. @Swati4star @vyankatesh24 Please help.

Has the problem resolved?

**Comment by jddeep** [comment]
> Has the problem resolved?

@Swati4star @vyankatesh24 I could not test that due to 
Screenshot (165)

File: app/src/main/res/layout/fragment_excelto_pdf.xml (hunk 18 above)
Comment (jddeep) [reply app/src/main/res/layout/fragment_excelto_pdf.xml]: will removw it from push

File: app/build.gradle (hunk 2 above)
Comment (Swati4star) [reply app/build.gradle]: @jddeep ?

File: app/build.gradle [hunk 20]
```diff
@@ -49,7 +59,8 @@ dependencies {
     implementation 'com.android.support:cardview-v7:28.0.0'
     implementation 'com.android.support:support-v4:28.0.0'
     implementation 'com.android.support:preference-v7:28.0.0'
-
+//    implementation 'com.aspose:aspose-cells:19.3'
```
Comment (Swati4star) [inline app/build.gradle]: no need of this

File: app/src/main/res/layout/fragment_excelto_pdf.xml (hunk 18 above)
Comment (Swati4star) [reply app/src/main/res/layout/fragment_excelto_pdf.xml]: not yet removed

File: app/build.gradle (hunk 2 above)
Comment (jddeep) [reply app/build.gradle]: Yes @Swati4star It is required for the aspose and grapecity library implementations and working.

Comment (jddeep) [reply app/build.gradle]: And the packaging options were provided because Android Gradle builds, you're not permitted to include the same file with the same path more than once in the output. In your build, there were two META-INF/DEPENDENCIES files coming from different places. This problem was in the Zip to Pdf fragment,

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java (hunk 13 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: @Swati4star I have tried that in this push. That error seems to go away now.

**Comment by vyankatesh24** [comment]
@jddeep Have a look checks are failing.

**Comment by jddeep** [comment]
@vyankatesh24 Please give me 15 mins. I will be looking into it. I am having some issue with my VCS and gitbash.

**Comment by jddeep** [comment]
@Swati4star @vyankatesh24 The build still fails, I updated the configurations which worked so far but still the build fails, Any suggestions?

**Comment by vyankatesh24** [comment]
> @Swati4star @vyankatesh24 The build still fails, I updated the configurations which worked so far but still the build fails, Any suggestions?

 ./gradlew assembleDebug assembleRelease
 ./gradlew checkstyle

Run these commands locally in android studio terminal
you will get to see what the error is

**Comment by jddeep** [comment]
@Swati4star Please review now.

**Comment by vyankatesh24** [comment]
@jddeep 
Still, I'm getting `Error occured` snackbar when clicked on `Excel to PDF`

File: app/src/main/res/layout/fragment_excelto_pdf.xml [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+            android:text="Select a Excel file" />
+        <com.dd.morphingbutton.MorphingButton
+            android:id="@+id/create_excel_to_pdf"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            style="@style/MorphingButton"
+            android:layout_gravity="center"
+            android:layout_marginTop="10dp"
+            android:padding="10dp"
+            android:text="Excel to pdf" />
```
Comment (vyankatesh24) [inline app/src/main/res/layout/fragment_excelto_pdf.xml]: Use String.xml

File: app/src/main/res/layout/fragment_excelto_pdf.xml [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+        android:layout_margin="@dimen/activity_horizontal_margin"
+        android:orientation="vertical">
+
+        <com.dd.morphingbutton.MorphingButton
+            android:id="@+id/select_excel_file"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            style="@style/MorphingButton"
+            android:padding="10dp"
+            android:text="Select a Excel file" />
```
Comment (vyankatesh24) [inline app/src/main/res/layout/fragment_excelto_pdf.xml]: Use string from `string.xml`

File: app/src/main/res/menu/activity_main_drawer.xml [hunk 23]
```diff
@@ -20,6 +20,9 @@
                         android:id="@+id/nav_qrcode"
                         android:icon="@drawable/ic_qrcode_24dp"
                         android:title="@string/qr_barcode_pdf" />
+                    <item
+                        android:id="@+id/nav_excel_to_pdf"
+                        android:title="@string/excel_to_pdf" />
```
Comment (vyankatesh24) [inline app/src/main/res/menu/activity_main_drawer.xml]: Add appropriate icon
Also change text in navigation drawer from `Excel to Pdf` to `Excel to PDF`.

File: app/src/main/res/values/strings.xml [hunk 24]
```diff
@@ -122,6 +122,8 @@
     <string name="snackbar_file_not_deleted">File can\'t be deleted.</string>
     <string name="snackbar_folder_not_created">Error on creating application folder.</string>
     <string name="pdf_merged">PDFs Merged</string>
+    <string name="excel_selected">Excel file selected: </string>
```
Comment (vyankatesh24) [inline app/src/main/res/values/strings.xml]: You are showing this string on snackbar, so remove `:` from this string

**Comment by jddeep** [comment]
> @jddeep
> Still, I'm getting `Error occured` snackbar when clicked on `Excel to PDF`

@vyankatesh24 But here i am not getting the snackbar error! I will further confirm again in the afternoon once. And also you need to do include the aspose/grapecity jar libraries along with some more steps. I will send you the documentation link in the afternoon. It will work only after that.

File: app/src/main/res/values/strings.xml (hunk 24 above)
Comment (jddeep) [reply app/src/main/res/values/strings.xml]: Sure

File: app/src/main/res/menu/activity_main_drawer.xml (hunk 23 above)
Comment (jddeep) [reply app/src/main/res/menu/activity_main_drawer.xml]: Okay

**Comment by jddeep** [comment]
@Swati4star @vyankatesh24 Please refer to here https://docs.aspose.com/display/cellsjava/Aspose.Cells+for+Android+via+Java+Installation to test it out .
And yes @vyankatesh24 Everything works fine and there is no error. I will be updating the PR template with a gif. All asked formatting changes have also been made. Please review and merge. Thanks!

**Comment by Swati4star** [comment]
@vyankatesh24 Please review it finally.

**Comment by vyankatesh24** [comment]
@jddeep 

Still, I'm not able to create PDF successfully.
the error are as follows:


1)

`java.lang.IllegalArgumentException: permission cannot be null or empty
W/System.err:     at android.content.pm.PackageManager.buildRequestPermissionsIntent(PackageManager.java:3465)
        at android.app.Activity.requestPermissions(Activity.java:4292)
        at android.support.v4.app.ActivityCompat.requestPermissions(ActivityCompat.java:502)
        at swati4star.createpdf.util.PermissionsUtils.requestRuntimePermissions(PermissionsUtils.java:60)
        at swati4star.createpdf.util.PermissionsUtils.checkRuntimePermissions(PermissionsUtils.java:38)
        at swati4star.createpdf.fragment.ExceltoPdfFragment.getRuntimePermissions(ExceltoPdfFragment.java:214)
        at swati4star.createpdf.fragment.ExceltoPdfFragment.openExcelToPdf(ExceltoPdfFragment.java:110)
        at swati4star.createpdf.fragment.ExceltoPdfFragment_ViewBinding$1.doClick(ExceltoPdfFragment_ViewBinding.java:35)
        at butterknife.internal.DebouncingOnClickListener.onClick(DebouncingOnClickListener.java:22)
W/System.err:     at android.view.View.performClick(View.java:5647)`




2)

`System.err: java.lang.NullPointerException: File path can't be null.
        at com.aspose.cells.c.a.d.ze.a(Unknown Source)
W/System.err:     at com.aspose.cells.c.a.d.ze.<init>(Unknown Source)
        at com.aspose.cells.c.a.d.zd.a(Unknown Source)
        at com.aspose.cells.Workbook.a(Unknown Source)
        at com.aspose.cells.Workbook.<init>(Unknown Source)
        at swati4star.createpdf.fragment.ExceltoPdfFragment.convertToPdf(ExceltoPdfFragment.java:195)
        at swati4star.createpdf.fragment.ExceltoPdfFragment.lambda$openExcelToPdf$2(ExceltoPdfFragment.java:122)
        at swati4star.createpdf.fragment.-$$Lambda$ExceltoPdfFragment$YHulokh4hmJIPgynrKbknr3qjlM.onInput(lambda)
        at com.afollestad.materialdialogs.MaterialDialog.onClick(MaterialDialog.java:419)`


@sidhuparas @srv-twry can you please try testing on your phone. (the second error could be only on my device)

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java [hunk 25]
```diff
@@ -0,0 +1,221 @@
...
+                        if (fileName.endsWith(Constants.excelExtension))
+                            mFileExtension = Constants.excelExtension;
+                        else if (fileName.endsWith(Constants.excelWorkbookExtension))
+                            mFileExtension = Constants.excelWorkbookExtension;
+                        else {
+                            showSnackbar(mActivity, R.string.extension_not_supported);
+                            return;
+                        }
+                    }
+                    fileName = getResources().getString(R.string.excel_selected)
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: I think here it should be `excel_tv_view_text` rather than `R.string.excel_selected`

**Comment by jddeep** [comment]
@vyankatesh24 I think first error is Permission request related. It is possible as the Excel to PDF is a fragment. Though I didn't get that error while testing as you can see from the shared gif. But I could not get the second error everything should works perfectly. @Swati4star Can you please see into this.

**Comment by jddeep** [comment]
And for fixing the Permission issue (though I didn't get it on my device while testing) is a issue for requests in Fragments I would like to start working on #669 . @Swati4star

File: app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java (hunk 25 above)
Comment (jddeep) [reply app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java]: Okay will change that

**Comment by jddeep** [comment]
> Also you haven't yet added an option on `Home Fragment`

@vyankatesh24 because in #186 It wasn't supposed to be added. I followed the issue requirements there.

**Comment by jddeep** [comment]
@vyankatesh24 @Swati4star Please finalize this today and merge.

**Comment by jddeep** [comment]
Please follow setup instructions from here https://docs.aspose.com/display/cellsjava/Aspose.Cells+for+Android+via+Java+Installation @vyankatesh24 . I am mailing you the library jar file and the aspose apk.

**Comment by vyankatesh24** [comment]
> Also you haven't yet added an option on `Home Fragment`

@jddeep You haven't taken care of this yet.

**Comment by jddeep** [comment]
> > Also you haven't yet added an option on `Home Fragment`
> 
> @vyankatesh24 because in #186 It wasn't supposed to be added. I followed the issue requirements there.

@vyankatesh24 Please see this. If you still want I shall add it.

**Comment by vyankatesh24** [comment]
@Swati4star 
Also, the `Aspose` library @jddeep using is on 30 days trial only. 
@jddeep correect 

> > > Also you haven't yet added an option on `Home Fragment`
> > 
> > 
> > @vyankatesh24 because in #186 It wasn't supposed to be added. I followed the issue requirements there.
> 
> @vyankatesh24 Please see this. If you still want I shall add it.

It will be good if you do!

**Comment by jddeep** [comment]
> Also, the Aspose library @jddeep using is on 30 days trial only.
> @jddeep correect

@vyankatesh24 You are talking about the full Aspose API. Here I have used only the jar library locally which is free. If I was to use the API then I didn't need to include the library locally. Plus they give some more features with the API.

**Comment by jddeep** [comment]
@vyankatesh24 I will be adding this in the Home Fragment also. Can you tell me the position where to add.

**Comment by vyankatesh24** [comment]
@jddeep 
Great Job :+1: 
I do tested by directly giving file path (in code) and it's working.

**Comment by vyankatesh24** [comment]
> > Also, the Aspose library @jddeep using is on 30 days trial only.
> > @jddeep correect
> 
> @vyankatesh24 You are talking about the full Aspose API. Here I have used only the jar library locally which is free. If I was to use the API then I didn't need to include the library locally. Plus they give some more features with the API.

Ok! if it's free then no problem :+1:

**Comment by jddeep** [comment]
Okay @vyankatesh24 I am updating the Home Fragment and pinging you on finishing that. Let's get this awesome feature merged.

**Comment by vyankatesh24** [comment]
> @vyankatesh24 I will be adding this in the Home Fragment also. Can you tell me the position where to add.

In `Create new PDF` section.

**Comment by vyankatesh24** [comment]
> Okay @vyankatesh24 I am updating the Home Fragment and pinging you on finishing that. Let's get this awesome feature merged.

Also one little change
change `SELECT A FILE TO UPLOAD` to `SELECT EXCEL FILE`
Rest everything looks good :+1:

**Comment by jddeep** [comment]
@vyankatesh24 @Swati4star Please merge. Thank you.

**Comment by vyankatesh24** [comment]
@jddeep I have approved the changes.
Well Done :+1: 
Just resolve the conflicts then I will merge it.

**Comment by jddeep** [comment]
@vyankatesh24 Conflicts Resolved! Please merge. Thanks
